proto/gnmi_ext/gnmi_ext.proto

```diff
@@ -89,3 +92,38 @@ message TimeRange {
   int64 start = 1; // Nanoseconds since the epoch
   int64 end = 2;   // Nanoseconds since the epoch
 }
+
+// Commit confirmed extension allows automated revert of a configuration after
+// certain duration if an explicit confirmation is not issued. This duration can
+// be used to verify the configuration.
+// It also allows explicit cancellation of the commit during the time window of rollback
+// duration if client determine that the configuration needs to be reverted.
+// The document about gNMI commit confirmed can be found at
+// https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-commit-confirmed.md
+message Commit {
+  oneof action {
+    CommitRequest commit = 1;
+    CommitConfirm confirm = 2;
+    CommitCancel cancel = 3;
+  }
+}
+
+// Create a new commit request.
+message CommitRequest {
+  // Commit id for the request.
+  string id = 1;
```

Comment:
@kidiyoor what are the semantics behind this `id`?
Is it returned by the server when CommitRequest is issued?

Reply:
It is passed by the client making the commit request. If the id is not passed its a bad request.

Reply:
can i use gnmic tool to request for gnmi commit ?